plugins/cue_points/quiz/services/QuizrReportingService.php

```diff
@@ -0,0 +1,133 @@
+<?php
+/**
+ * Allows user to get reports regarding quizes
+ *
+ * @service quizReports
+ * @package plugins.quiz
+ * @subpackage api.services
+ */
+
+class QuizrReportingService extends KalturaBaseService
+{
+
+	public function initService($serviceId, $serviceName, $actionName)
+	{
+		parent::initService($serviceId, $serviceName, $actionName);
+
+		if(!QuizPlugin::isAllowedPartner($this->getPartnerId()))
```

Comment:
@MosheMaorKaltura this change will be incorrect as $this->partnerId is initialized in parent::initService